polkadot/runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -50,3 +50,8 @@ pub fn backing_constraints<T: initializer::Config>(
 pub fn scheduling_lookahead<T: initializer::Config>() -> u32 {
 	configuration::ActiveConfig::<T>::get().scheduler_params.lookahead
 }
+
+/// Implementation for `validation_code_bomb_limit` function from the runtime API
+pub fn validation_code_bomb_limit<T: initializer::Config>() -> u32 {
```

Comment:
With this, you slightly couple runtime upgrade to every node already upgrading to use this new runtime call. If we want complete control when the new limit is used I would simply just pick a feature flag for it and write it like this:
NewCodeLimit

```
if DYNAMIC_CODE_BOMB_LIMIT set {
   configuration::ActiveConfig::<T>::get().max_code_size * configuration::MAX_VALIDATION_CODE_COMPRESSION_RATIO
} else {
    polkadot_node_primitives::VALIDATION_CODE_BOMB_LIMIT
}
```

The disavantage is that you need an extra referenda.

Reply:
Yeah, the intention is to do this fix ASAP. The runtime upgrade will enable the new limit.